[2025-06-18T03:01:42.170+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-06-18T03:01:42.195+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_read_from_bigquery.read_bq manual__2025-06-18T03:01:41.045817+00:00 [queued]>
[2025-06-18T03:01:42.205+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_read_from_bigquery.read_bq manual__2025-06-18T03:01:41.045817+00:00 [queued]>
[2025-06-18T03:01:42.206+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-06-18T03:01:42.223+0000] {taskinstance.py:2890} INFO - Executing <Task(SparkSubmitOperator): read_bq> on 2025-06-18 03:01:41.045817+00:00
[2025-06-18T03:01:42.230+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_read_from_bigquery', 'read_bq', 'manual__2025-06-18T03:01:41.045817+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/test_scipt/test_read_bq.py', '--cfg-path', '/tmp/tmpdryul3c5']
[2025-06-18T03:01:42.232+0000] {standard_task_runner.py:105} INFO - Job 4: Subtask read_bq
[2025-06-18T03:01:42.234+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=2169) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-06-18T03:01:42.234+0000] {standard_task_runner.py:72} INFO - Started process 2170 to run task
[2025-06-18T03:01:42.274+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_read_from_bigquery.read_bq manual__2025-06-18T03:01:41.045817+00:00 [running]> on host 9a450f683bb5
[2025-06-18T03:01:42.372+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='you' AIRFLOW_CTX_DAG_ID='spark_read_from_bigquery' AIRFLOW_CTX_TASK_ID='read_bq' AIRFLOW_CTX_EXECUTION_DATE='2025-06-18T03:01:41.045817+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-18T03:01:41.045817+00:00'
[2025-06-18T03:01:42.373+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-06-18T03:01:42.403+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2025-06-18T03:01:42.404+0000] {spark_submit.py:351} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --packages com.google.cloud.spark:spark-bigquery-with-dependencies_2.12:0.42.2 --name spark_read_bq_job /opt/bitnami/spark/app/test_data_bq.py
[2025-06-18T03:01:43.805+0000] {spark_submit.py:521} INFO - :: loading settings :: url = jar:file:/opt/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-06-18T03:01:43.899+0000] {spark_submit.py:521} INFO - Ivy Default Cache set to: /home/***/.ivy2/cache
[2025-06-18T03:01:43.899+0000] {spark_submit.py:521} INFO - The jars for the packages stored in: /home/***/.ivy2/jars
[2025-06-18T03:01:43.904+0000] {spark_submit.py:521} INFO - com.google.cloud.spark#spark-bigquery-with-dependencies_2.12 added as a dependency
[2025-06-18T03:01:43.905+0000] {spark_submit.py:521} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-947d94ea-0a2b-4113-a9d0-c2423c1e0985;1.0
[2025-06-18T03:01:43.906+0000] {spark_submit.py:521} INFO - confs: [default]
[2025-06-18T03:01:44.060+0000] {spark_submit.py:521} INFO - found com.google.cloud.spark#spark-bigquery-with-dependencies_2.12;0.42.2 in central
[2025-06-18T03:01:44.075+0000] {spark_submit.py:521} INFO - :: resolution report :: resolve 166ms :: artifacts dl 4ms
[2025-06-18T03:01:44.076+0000] {spark_submit.py:521} INFO - :: modules in use:
[2025-06-18T03:01:44.076+0000] {spark_submit.py:521} INFO - com.google.cloud.spark#spark-bigquery-with-dependencies_2.12;0.42.2 from central in [default]
[2025-06-18T03:01:44.077+0000] {spark_submit.py:521} INFO - ---------------------------------------------------------------------
[2025-06-18T03:01:44.077+0000] {spark_submit.py:521} INFO - |                  |            modules            ||   artifacts   |
[2025-06-18T03:01:44.078+0000] {spark_submit.py:521} INFO - |       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-06-18T03:01:44.078+0000] {spark_submit.py:521} INFO - ---------------------------------------------------------------------
[2025-06-18T03:01:44.078+0000] {spark_submit.py:521} INFO - |      default     |   1   |   0   |   0   |   0   ||   1   |   0   |
[2025-06-18T03:01:44.079+0000] {spark_submit.py:521} INFO - ---------------------------------------------------------------------
[2025-06-18T03:01:44.079+0000] {spark_submit.py:521} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-947d94ea-0a2b-4113-a9d0-c2423c1e0985
[2025-06-18T03:01:44.079+0000] {spark_submit.py:521} INFO - confs: [default]
[2025-06-18T03:01:44.082+0000] {spark_submit.py:521} INFO - 0 artifacts copied, 1 already retrieved (0kB/4ms)
[2025-06-18T03:01:44.294+0000] {spark_submit.py:521} INFO - 25/06/18 03:01:44 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-06-18T03:01:44.728+0000] {spark_submit.py:521} INFO - python3: can't open file '/opt/bitnami/spark/app/test_data_bq.py': [Errno 2] No such file or directory
[2025-06-18T03:01:44.742+0000] {spark_submit.py:521} INFO - 25/06/18 03:01:44 INFO ShutdownHookManager: Shutdown hook called
[2025-06-18T03:01:44.743+0000] {spark_submit.py:521} INFO - 25/06/18 03:01:44 INFO ShutdownHookManager: Deleting directory /tmp/spark-95d9f722-fba9-41ea-9ac6-1ed1873a1d68
[2025-06-18T03:01:44.784+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 452, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --packages com.google.cloud.spark:spark-bigquery-with-dependencies_2.12:0.42.2 --name spark_read_bq_job /opt/bitnami/spark/app/test_data_bq.py. Error code is: 2.
[2025-06-18T03:01:44.797+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=spark_read_from_bigquery, task_id=read_bq, run_id=manual__2025-06-18T03:01:41.045817+00:00, execution_date=20250618T030141, start_date=20250618T030142, end_date=20250618T030144
[2025-06-18T03:01:44.825+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-06-18T03:01:44.826+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 4 for task read_bq (Cannot execute: spark-submit --master spark://spark-master:7077 --packages com.google.cloud.spark:spark-bigquery-with-dependencies_2.12:0.42.2 --name spark_read_bq_job /opt/bitnami/spark/app/test_data_bq.py. Error code is: 2.; 2170)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 452, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --packages com.google.cloud.spark:spark-bigquery-with-dependencies_2.12:0.42.2 --name spark_read_bq_job /opt/bitnami/spark/app/test_data_bq.py. Error code is: 2.
[2025-06-18T03:01:44.859+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-06-18T03:01:44.873+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-06-18T03:01:44.875+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
